src/sources/splunk_hec.rs

```diff
@@ -1056,7 +1056,7 @@ mod tests {
     async fn partial() {
         trace_init();
 
-        let message = r#"{"event":"first"}{"event":"second""#;
+        let message = r#"{"event":"first"}{"event":"second"}{"event": ""#;
```

Comment:
I think this test breaking here is a sign we have done something wrong. This change seems to invalidate the assertion of the test?

Reply:
Yeah, exactly :smile: I'm pretty sure there was an issue introduced in #4875 where it won't forward a single event until another is published. I was hoping you might be able to take a look.

Apologies, I could have made it clearer in the PR description that I don't think this is the right fix. I pushed it just for demonstration purposes.

Reply:
Ah ok, let me investigate.